### PR TITLE
[feature] Pivot node validation

### PR DESCRIPTION
**Pull request by re1ro**
Adding validation to a pivot type node, so we don't send empty ranges query to ES

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 1b248af4e5ee733a3002d08bdca7db2e6b6b7495


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bb7ca4a1517f7192fc1dc8ef11fc5526ffdf8aed
</p>

File: src/exampleTypes.js [hunk 1]
```diff
@@ -263,6 +263,14 @@ export default F.stampKey('type', {
     },
   },
   pivot: {
+    validate: context =>
+      _.every(
+        ({ type, ranges, percents }) =>
+          (type !== 'numberRanges' && type !== 'percentiles') ||
+          (type === 'numberRanges' && ranges.length > 0) ||
+          (type === 'percentiles' && percents.length > 0),
```
Comment (daedalus28) [inline src/exampleTypes.js:271]: Do these throw an error? Maybe we should actually do this in contexture-es 🤔

Comment (re1ro) [reply src/exampleTypes.js:271]: Yeah ES returns an error if it's empty

Comment (daedalus28) [reply src/exampleTypes.js:271]: OK, merging for now but we should fix this at the contexture-es level.

For a future revision, this kind of thing might be more cleanly modeled with an object lookup for type specific checks, e.g. something like this:

```js
let checks = {
  numberRanges: ({ ranges }) => ranges.length > 0,
  percentiles: ({ percents }) => percents.length > 0,
}
let checkGroup = group => _.getOr(_.stubTrue, group.type, checks)(group)
let validate = ({ groups }) => _.every(checkGroup, groups)
```

**Pull request merged**